cmd/test/makefiles/speed.go

```diff
@@ -0,0 +1,245 @@
+package makefiles
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"time"
+
+	"github.com/rclone/rclone/cmd"
+	"github.com/rclone/rclone/cmd/test"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/cache"
+	"github.com/rclone/rclone/fs/config/flags"
+	"github.com/rclone/rclone/fs/operations"
+	"github.com/rclone/rclone/fs/sync"
+	"github.com/rclone/rclone/lib/atexit"
+	"github.com/rclone/rclone/lib/random"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// Flags
+	testTime = fs.Duration(15 * time.Second)
+	fcap     = 100
+	small    = fs.SizeSuffix(1024)
+	medium   = fs.SizeSuffix(10 * 1024 * 1024)
+	large    = fs.SizeSuffix(1024 * 1024 * 1024)
+	useJSON  = false
+)
+
+func init() {
+	test.Command.AddCommand(speedCmd)
+
+	speedFlags := speedCmd.Flags()
+	flags.FVarP(speedFlags, &testTime, "test-time", "", "Length for each test to run", "")
+	flags.IntVarP(speedFlags, &fcap, "file-cap", "", fcap, "Maximum number of files to use in each test", "")
+	flags.FVarP(speedFlags, &small, "small", "", "Size of small files", "")
+	flags.FVarP(speedFlags, &medium, "medium", "", "Size of medium files", "")
+	flags.FVarP(speedFlags, &large, "large", "", "Size of large files", "")
+	flags.BoolVarP(speedFlags, &useJSON, "json", "", useJSON, "Output only results in JSON format", "")
+
```

Comment:
Can you factor this bit of code into a function - say `addCommonFlags`

https://github.com/rclone/rclone/blob/0019e18ac3a673714e851b2a97d63a8ad0a77654/cmd/test/makefiles/makefiles.go#L64-L72

Then call it here `addCommonFlags(speedFlags)` so this command has the flags for controlling the contents of the test files too.